src/middleware/errorMiddleware.test.ts

```diff
@@ -0,0 +1,21 @@
+import StatusError from '@util/error';
+import { NextFunction, Request, Response } from 'express';
+import { errorMiddleware } from './errorMiddleware';
+
+const req = {} as Request;
+const res = {} as Response;
+const next = jest.fn() as NextFunction;
+const statusError = new StatusError(404, '');
+
+beforeAll(() => {
+  res.status = jest.fn().mockImplementation(() => res);
```

Comment:
aqui pode ser apenas jest.fn(), pois isso já é uma função mockada, além de dever estar num objeto mockado e não dentro de um beforeAll.

Uma dica para saber quando utilizar o beforeAll / each é quando vc ter vários testes e o mesmo código repetir-sem em vários.

Reply:
Removi o beforeAll, todavia não é possível fazer o valor de res.status ser apenas jest.fn() pois é importante que ele retorne uma Response de forma que seja possível dar comandos em seguida.
Por exemplo:
```
// Com mais que jest.fn()
return res.status(200).json({}) -> return res.json({}) -> return res

// Apenas jest.fn()
return res.status(200).json({}) -> return undefined.json({}) -> ERRO
```

Reply:
Você pode mockar o retorno do status pra retornar o this, sendo assim possível utilizar status.json:

```
const res = { 
     status: jest.fn().mockReturnThis(),
     json: jest.fn()
} 
  ```

Reply:
muito obrigado, feito!